system/system.proto

```diff
@@ -227,6 +232,86 @@ message PingResponse {
   int32 ttl = 13;           // Remaining time to live value.
 }
 
+// A ProbeRequest describes the probe operation to perform.  The
+// destination field and exactly one of the probe_target fields
+// are required.  Any field not specified is set to a reasonable server
+// specified value.  Not all fields are supported by all vendors.
+message ProbeRequest {
+  string destination = 1;   // Proxy address to probe. required.
+  string source = 2;        // Source address to probe from.
+  oneof probe_target {
+    // exactly one of the next 4 items must be present.
+    int32 ifindex = 3;        // ifIndex to probe, if present.
+    string ifname = 4;        // ifName to probe, if present.
+    string ifaddr = 5;        // ifAddr to probe, if present.
+    string remote = 6;        // remote address to probe, if present.
+  }
+  int32 count = 7;          // Number of requests to make, default 3.
+  int64 interval = 8;       // Nanoseconds between requests.
+  int64 wait = 9;           // Nanoseconds to wait for a response.
+
+  string network_instance = 10; // Network instance to probe the proxy node in.
+}
+
+// A ProbeResponse represents either the response to a single probe packet
+// (the code field is non-zero) or the summary statistics (sent is non-zero).
+//
+// For a single probe packet, time is the round trip time, in nanoseconds.  For
+// summary statistics, it is the time spent by the ping operation.  The time is
+// not always present in summary statistics.
+//
+// In the summary, each map is keyed by the enum values referred to by that map.
+// The codes map is keyed by the Code enum, the states map is keyed by the State
+// enum, and the actives map is keyed by the Active enum.
+message ProbeResponse {
+  enum Code {
+    CODE_UNKNOWN = 0;
+    CODE_NO_ERROR = 1;
+    CODE_MALFORMED_QUERY = 2;
+    CODE_NO_SUCH_INTERFACE = 3;
+    CODE_NO_SUCH_TABLE_ENTRY = 4;
+    CODE_MULTIPLE_INTERFACES = 5;
+  }
+  enum State {
+    STATE_UNKNOWN = 0;
```

Comment:
`STATE_RESERVED` to follow the rfc

Reply:
I think its fine to have this as `UNKNOWN`. If we want to have a reserved value then we should probably have a different value that doesn't have semantics associated with it. proto style is for this to be `UNKNOWN`/`UNSPECIFIED` rather than have semantic meaning.

Reply:
Thanks, Rob!